Diary refactor

andrewdwooten opened the pull request:
Extracts functions from diary.js into diary-food-add.js,
diary-food-remove.js, diary-food-search.js, and diary-food-table.js.
Functions are logically broken into each of the remainging files in
the diary folder. Necessary exports and changes to function
invocation were performed in order to maintain proper functionality.
All page behavior remains appropriate to specifications.

andrewdwooten committed:
Extract functions to diary-food-table

Extracts getFoods and makeFoodTable function to diary-food-table.js.
Modifies diary.js to call appropriate function from exports on
document ready. All functionality is as prior to refactor.

andrewdwooten committed:
Extract calorie functions
Extracts all calorie frunctions from diary.js to calories.js.
Modifies all function invocations in diary.js to appropriately
call module exports from calories.js. All page functionality
remains as it was prior to refactor.

andrewdwooten committed:
Extract food addition functions to diary-food-add.js

Extracts all functions regarding food addition to diary-food-add.js.
Functions added to module.exports for this file. Necessary changes made
to diary.js. All page behavior is appropriate at this time.

andrewdwooten committed:
extract food removal functions to diary-food-remove

Extracts all food removal functions to diary-food-remove.js.
Necessary changes made to diary.js to allow for appropriate function
of app features. All page behavior is normal.

andrewdwooten committed:
Complete logic extraction from diary

Extracts functions from diary.js into diary-food-add.js,
diary-food-remove.js, diary-food-search.js, and diary-food-table.js.
Functions are logically broken into each of the remainging files in
the diary folder. Necessary exports and changes to function
invocation were performed in order to maintain proper functionality.
All page behavior remains appropriate to specifications.

CjMoore commented on lib/diary/diary.js:
```diff
@@ -0,0 +1,147 @@
+  } else {
+    $('#date-box').attr('data-diary-id', `${data.id}`)
+  }
+}
```
I think we should move this within the class.

CjMoore commented on lib/diary/diary.js:
```diff
@@ -0,0 +1,147 @@
+  const diary = new Diary
+
+  diaryFoodTable.getFoods()
+
```
we can refactor this further and add a diary listeners function within the diary. We can then call it in the constructor.

andrewdwooten committed:
populateMealsTables moved withing diary class

andrewdwooten committed:
Adds requested refactors

Adds contructor to diary class which calls getDiary
and new diary listeners function on construction.
diaryListeners function is an instance method for
the diary class which adds event listeners for
navigation between diary dates. Also moves
diaryFoodTable.getFoods invocation to construction
of diary(this triggers the building of the foods
table on the diary page). All page functionality
remains as before.

andrewdwooten committed:
update jquery requires and runs npm build

CjMoore reviewed (approved):
👍
